soc/nxp/imx/imx9/imx95/m7/soc.c

```diff
@@ -214,6 +222,34 @@ void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 	__enable_irq();
 }
 
+#if defined(CONFIG_REBOOT)
+void sys_arch_reboot(int type)
+{
+	struct scmi_system_power_state_config cfg;
```

Comment:
This looks identical with the function for imx943. Is there a way to avoid duplicating code?